filters/fadein/fadein.go

```diff
@@ -0,0 +1,260 @@
+package fadein
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/routing"
+)
+
+const (
+	FadeInName          = "fadeIn"
+	EndpointCreatedName = "endpointCreated"
+)
+
+type (
+	fadeIn struct {
+		duration time.Duration
+		exponent float64
+	}
+
+	endpointCreated struct {
+		when  time.Time
+		which string
+	}
+
+	detectedFadeIn struct {
+		when       time.Time
+		duration   time.Duration
+		lastActive time.Time
+	}
+
+	postProcessor struct {
+		// "http://10.2.1.53:1234": {t0 60s t0-10s}
+		detected map[string]detectedFadeIn
+	}
+)
+
+// NewFadeIn creates a filter spec for the fade-in filter.
+func NewFadeIn() filters.Spec {
+	return fadeIn{}
+}
+
+func (fadeIn) Name() string { return FadeInName }
+
+func (fadeIn) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) == 0 || len(args) > 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	var f fadeIn
+	switch v := args[0].(type) {
+	case int:
+		f.duration = time.Duration(v * int(time.Millisecond))
+	case float64:
+		f.duration = time.Duration(int(v) * int(time.Millisecond))
```

Comment:
`int(v)` truncates the float64 value. See also https://play.golang.org/p/3wFYVtpac9j
You want this:
```
	time.Duration(int(v * float64(time.Millisecond)))
```